clients/src/main/java/org/apache/kafka/clients/admin/NewPartitionReassignment.java

```diff
@@ -28,17 +28,17 @@
  * A new partition reassignment, which can be applied via {@link AdminClient#alterPartitionReassignments(Map, AlterPartitionReassignmentsOptions)}.
  */
 public class NewPartitionReassignment {
-    private final List<Integer> targetBrokers;
+    private final List<Integer> targetReplicas;
 
-    public static Optional<NewPartitionReassignment> of(Integer... brokers) {
-        return Optional.of(new NewPartitionReassignment(Arrays.asList(brokers)));
+    public static Optional<NewPartitionReassignment> of(Integer... replicas) {
```

Comment:
Why do we return `Optional` here? Doesn't makes sense just by itself, unless some bigger picture requires it.

Reply:
In addition to this,  this way of taking vararg allows a user to pass `null` (and this code will throw NPE). Normally these methods are declared like:

`of(Integer replica, Integer... rest)` to make sure that invalid values can be passed in.

Reply:
I agree. I think this was done because of all of the tests need a `Optional<NewPartitionReassignment>`.

I change the method's signature and fixed all of the test that used this method.

Reply:
> of(Integer replica, Integer... rest) to make sure that invalid values can note be passed in.

With this you can sill do `of(null)`, no? This would throw a `NPE` in `Arrays.asList`.

Reply:
Thanks for fixing the tests. Shall we change the method parameter to: `of(Integer replica, Integer... rest)` instead, so that someone cannot call `of(null)`?

Reply:
Do we need this method at all? It doesn't seem to provide much value. Also, should we raise an error if the list is empty?

Reply:
> Do we need this method at all? It doesn't seem to provide much value. Also, should we raise an error if the list is empty?

This depends if we want to support this UX
```
NewPartitionReassignment.of(1, 2, 3)
```
vs
```
new NewPartitionReassignment(Arrays.asList(1, 2, 3))
```

I am going to remove this method. We can always add it later.

Reply:
>  Also, should we raise an error if the list is empty?

I don't know. Should this be enforce by the admin client (by the constructor of this class) or by the broker when the user attempts to make the assignment? There a few other reasons for which the broker may reject the assignment. Do we want to check this particular case on the client?

I vote for leaving it as is and validating this and other cases in the controller/broker.

Reply:
Oops, saw your comment after posting mine.

Yes, someone can still pass `null`, in one case it will be that the parameter `replica` is null, in other case it will be vararg array is null. I think the first one reflects what is actually going on. We should probably check for null anyway as NPE isn't nice.

The way we have now, someone can call `of` w/o any parameter and it will be empty list of `targetBrokers`. Do we support having empty list of `targetBrokers`?

We should handle NPE gracefully in any case.

Reply:
wow, I don't think we actually catch the case where the replica set is empty. Let me open a separate PR for this

Reply:
https://github.com/apache/kafka/pull/7574